# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234428281 18000
#      Thu Feb 12 03:44:41 2009 -0500
# Node ID 474f64ab905f6d7953a7d8177c96e86e5aa6d2d3
# Parent  d65a0a1780b68859d6902d316c779d06c90681dd
src/TEMPLATE-INST/Array-tc.cc: undo previous change

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,15 @@
 2009-02-12  John W. Eaton  <jwe@octave.org>
 
+	* TEMPLATE-INST/Array-tc.cc: Undo previous change.
+	(octave_sort<octave_value>::ascending_compare,
+	octave_sort<octave_value>::descending_compare):
+	Pass args by const reference instead of value.
+
 	* version.h (OCTAVE_VERSION): Now 3.1.52+.
 	(OCTAVE_API_VERSION): Now api-v34+.
 
 	* TEMPLATE-INST/Array-tc.cc
 	(octave_sort<octave_value>::ascending_compare,
 	octave_sort<octave_value>::descending_compare):
 	Delete unused template specializations.
 	Use NO_INSTANTIATE_ARRAY_SORT instead of INSTANTIATE_ARRAY_SORT
diff --git a/src/TEMPLATE-INST/Array-tc.cc b/src/TEMPLATE-INST/Array-tc.cc
--- a/src/TEMPLATE-INST/Array-tc.cc
+++ b/src/TEMPLATE-INST/Array-tc.cc
@@ -37,17 +37,34 @@ along with Octave; see the file COPYING.
 
 #include "DiagArray2.h"
 #include "DiagArray2.cc"
 
 #include "oct-obj.h"
 
 #include "oct-sort.cc"
 
-NO_INSTANTIATE_ARRAY_SORT (octave_value);
+// FIXME -- these comparisons don't look right.  Where do we sort
+// octave_value objects and expect them to be character strings?
+
+template <>
+bool
+octave_sort<octave_value>::ascending_compare (const octave_value& a, const octave_value& b)
+{
+  return (a.string_value () < b.string_value ());
+}
+
+template <>
+bool
+octave_sort<octave_value>::descending_compare (const octave_value& a, const octave_value& b)
+{
+  return (a.string_value () > b.string_value ());
+}
+
+INSTANTIATE_ARRAY_SORT (octave_value);
 
 INSTANTIATE_ARRAY (octave_value, OCTINTERP_API);
 
 template class OCTINTERP_API Array2<octave_value>;
 
 template class OCTINTERP_API ArrayN<octave_value>;
 
 /*
